access-token-management/src/AccessTokenManagement/Internal/DefaultClientCredentialsCacheKeyGenerator.cs

```diff
@@ -18,9 +20,17 @@ public ClientCredentialsCacheKey GenerateKey(
         ClientCredentialsClientName clientName,
         TokenRequestParameters? parameters = null)
     {
-        var scopePart = "s_" + parameters?.Scope;
-        var resourcePart = "r_" + parameters?.Resource;
+        var scopePart = "s_" + GetCacheKey(parameters?.Scope ?? string.Empty);
+        var resourcePart = "r_" + GetCacheKey(parameters?.Resource ?? string.Empty);
 
-        return ClientCredentialsCacheKey.Parse(options.Value.CacheKeyPrefix + clientName + "::" + scopePart + "::" + resourcePart);
+        return ClientCredentialsCacheKey.Parse(options.Value.CacheKeyPrefix + clientName + "::" + scopePart + "::" +
+                                               resourcePart);
+    }
+
+    private static string GetCacheKey(string value)
```

Comment:
Suggestions: `CreateHash(string? value) and short circuit if `string.IsNullEmptyOrWhiteSpace` returning `string.Empty` saving an hash function allocation.